RIPS/rip-9999.md

```diff
@@ -0,0 +1,313 @@
+ validationGasLimit,
+ commit, // 32 bytes hash
+ yParity, r, s
+```
+
+This transaction subtype extends the `DELEGATED_EOA_SUBTYPE` transactions with an ability to authorize some
+contract code to act as a Smart Contract Account and control the context of the EOA sender.
+
+The transaction signature (`yParity`, `r`, `s`) are interpreted as an ECDSA signature on the `secp256k1` curve over
+the message `keccak256(MAGIC || chainId || nonce || implementationAddress || commit)`.
+
+The `commit` value is a hash values that the implementation will validate.
+It is passed to the validation frame as part of the encoded transaction structure. 
+
+Note that account's `nonce` is not incremented when `AUTHORIZED_EOA_SUBTYPE` transaction is executed so the same
+authorization can be used unlimited number of times. In order to withdraw authorization, the account must perform
+any action that will increment its nonce.
+
+Transaction hash uniqueness is therefore guaranteed by a `bigNonce` field for the `NonceManager` contract.
+
+The main difference between `DELEGATED_EOA_SUBTYPE` and `AUTHORIZED_EOA_SUBTYPE` is that the transaction is handled
+as a full RIP-7560 Account Abstraction transaction.
+
+The `implementationAddress` will be invoked with the `validateTransaction` frame same as a regular
+Smart Contract Account would.
+This allows EOAs to temporarily transfer control of their assets to a Smart Contract Account but maintain an
+ability to revoke or override this authorisation.
+This also allows a cheap creation of AA accounts that do not require any code deployment, if the private key
+used to create the `yParity, r, s` signature does not exist.
+
+Note that most of the transaction fields are not signed by this transaction's signature.
+This means that the "implementation" contract is the entity responsible for implementing all kinds of protections,
+and the EOA account that signs an authorization fully cedes control to the "implementation" until
+the authorization is revoked.
+
+For the purposes of Gas Abstraction, the `paymasterData` parameter is not signed as well,
+so different Paymaster contracts can be used for subsequent authorized transactions using the same authorization.
+
+#### Using permanent authorizations to create Smart Contract Accounts
+
+This approach relies on the so-called "Nick's method", which includes deriving an address from some data and
+a randomized signature value, without ever knowing the ECDSA private key that would be able to control this address.
+
+Applying this approach to an `AUTHORIZED_EOA_SUBTYPE` transaction, a new account can be created,
+whose address is forever bound to `implementationAddress` and `commit`,
+without ever needing to deploy code for the contract.
+
+### Gas-sponsored time-limited code injection EOA transaction subtype
+
+```
+INSERT_CODE_EOA_SUBTYPE
+```
+
+[//]: # (TODO: 7377 probably needs codeAddr/storage simply because it has no deployment frame; we can have initCode)
```

Comment:
So is this something we should support? This is basically making a proxy contract enshrined in the protocol, which includes some changes to account structure and how EVM interacts with "aliased" accounts, so probably would require a separate EIP, wouldn't it?

Reply:
I don't think it requires changes to the account structure, nor should it affect how EVM interacts with these accounts.  In 7377 these are not changed.  It just makes the transition cheaper without becoming a DoS vector, since we know there is already an entry in the bytecodes db and the account can reference it.

OTOH, I don't see this as an important feature.  It's fine to include it as an optimization as long as it doesn't add complexity, but otherwise I'd drop it and use initCode.

Reply:
@yoavw @drortirosh please see the commit changing the logic from EIP-7377-like to legacy-like:

https://github.com/eth-infinitism/RIPs/pull/1/commits/f568e828858cef36ef73ab21999c0e8f562f77ef

Is this better?